src/services/utils.ts

```diff
@@ -41,6 +41,23 @@ export const sleep = (mills: number) => {
   })
 }
 
+export const personalizeStep = async (signer: JsonRpcSigner, step: Step): Promise<Step> => {
+  if (step.action.toAddress && step.action.fromAddress) return step
+
+  const address = await signer.getAddress()
+  const fromAddress = step.action.fromAddress || address
+  const toAddress = step.action.toAddress || address
+
+  return {
+    ...step,
+    action: {
+      fromAddress,
+      toAddress,
+      ...step.action,
+    },
+  }
```

Comment:
`...step.action` will overwrite `fromAddress` and `toAddress` because of the order in the new object. It should work here if `fromAddress` and `toAddress` are not present in `action` before, but if the are defined with a falsey value, they would overwrite the addresse we want to set. So we should change the order.

Reply:
äh ja totally right. The spread has to come first of course 👍